tests/ops/pulse/test_convenience_functions.py

```diff
@@ -0,0 +1,426 @@
+        def circuit(params):
+            qml.evolve(H)(params=params, t=t)
+            return qml.expval(qml.PauliZ(0) @ qml.PauliZ(1))
+
+        @jax.jit
+        @qml.qnode(dev, interface="jax")
+        def jitted_circuit(params):
+            qml.evolve(H)(params=params, t=t)
+            return qml.expval(qml.PauliZ(0) @ qml.PauliZ(1))
+
+        constants = np.linspace(0, 9, 10)
+        params = [constants, constants]
+
+        assert qml.math.allclose(jitted_circuit(params), circuit(params), atol=5e-3)
+        assert qml.math.allclose(
+            jax.grad(jitted_circuit)(params), jax.grad(circuit)(params), atol=5e-3
+        )
+
+    def test_parametrized_hamiltonian_with_pwc_from_function(self):
+        """Test that a function decorated by pwc_from_function can be used to create a ParametrizedHamiltonian"""
+
+        @qml.pulse.pwc_from_function((2, 5), 20)
+        def f1(params, t):
+            return params + t
+
+        @qml.pulse.pwc_from_function((3, 7), 10)
+        def f2(params, t):
+            return params[0] + params[1] * t**2
+
+        H = f1 * qml.PauliX(0) + f2 * qml.PauliY(1)
+        params = [1.2, [2.3, 3.4]]
+
+        assert isinstance(H, ParametrizedHamiltonian)
+
+        # at t=8 and t=1, both terms are 0
+        assert qml.math.allequal(qml.matrix(H(params, t=8)), 0)
+        assert qml.math.allequal(qml.matrix(H(params, t=1)), 0)
+
+        # at t=2.5, only term 1 is non-zero
+        true_mat = qml.matrix(f1(params[0], 2.5) * qml.PauliX(0), wire_order=[0, 1])
+        assert qml.math.allequal(qml.matrix(H(params, t=2.5)), true_mat)
+
+        # # at t=6, only term 2 is non-zero
+        true_mat = qml.matrix(f2(params[1], 6) * qml.PauliY(1), wire_order=[0, 1])
+        assert qml.math.allequal(qml.matrix(H(params, t=6)), true_mat)
+        #
+        # # at t=4, both are non-zero and output is as expected
+        true_mat = qml.matrix(f1(params[0], 4) * qml.PauliX(0), wire_order=[0, 1]) + qml.matrix(
+            f2(params[1], 4) * qml.PauliY(1), wire_order=[0, 1]
+        )
+        assert qml.math.allequal(qml.matrix(H(params, t=4)), true_mat)
+
+    @pytest.mark.slow
+    def test_qnode_pwc_from_function(self):
+        """Test that the evolution of a ParametrizedHamiltonian defined with a function decorated by pwc_from_function
+        can be executed on a QNode."""
+        import jax
+        import jax.numpy as jnp
+
+        @qml.pulse.pwc_from_function((2, 5), 20)
+        def f1(params, t):
+            return params + t
+
+        @qml.pulse.pwc_from_function((3, 7), 10)
+        def f2(params, t):
+            return params[0] + params[1] * t**2
+
+        H = f1 * qml.PauliX(0) + f2 * qml.PauliY(1)
+
+        t = (1, 4)
+
+        def generator(params):
+            time_step = 1e-3
+            times = jnp.arange(*t, step=time_step)
+            for ti in times:
+                yield jax.scipy.linalg.expm(-1j * time_step * qml.matrix(H(params, t=ti)))
```

Comment:
Since $X_0$ and $Y_1$ are commuting you can also compute the true time evolution simply as $\exp(-i F_1 X_0) \exp(-i F_2 Y_0)$ where $F_j = \int_1^4 d\tau f_j(p, \tau)$ - not sure that is the cause for the numerical discrepancy between the local and CI run.

Reply:
Might be better anyway to avoid unnecessarily expensive tests

Reply:
Thanks for the suggestion! This is much faster and fixes the failing test.